training/creating_dataset.py

```diff
@@ -73,12 +250,26 @@ def load_and_preprocess_data(
     # only npc monsters
     bestiary = bestiary[bestiary["type"] == "npc"]
     # only system column (all characteristics are there)
-    bestiary = bestiary.filter(regex="system", axis="columns")
+    # bestiary = bestiary.filter(regex="system", axis="columns")
+    (
+        speeds,
+        weaknesses,
+        resistances,
+        special_characteristics,
+        characteristics_rename,
+    ) = split_characteristics_into_groups(characteristics)
+
+    if "immunities" in characteristics_rename:
+        with pd.option_context("mode.chained_assignment", None):
+            immunities_path = CHARACTERISTICS_RENAME.get("immunities")
+            bestiary[immunities_path] = bestiary[immunities_path].apply(
+                lambda x: 0 if np.all(pd.isnull(x)) else len(x)
+            )
```

Comment:
1. To właściwie zlicza liczbę immunities, może warto byłoby to podkreślić? W sensie explicite cecha `num_immunities` czy jakoś tak.
2. Zrobiłbym zmienną z tej lambdy, i nadał jej nazwę, np. `count_immunities`.
3. Zamiast `np.all()` działa też zwykłe `sum()`, ale to bardziej dla waszej wiedzy, to jest czytelne. Chociaż czy zwykłe `if x` nie wystarcza?